### PR TITLE
refactor: 💡 update typings for MonkeyPatched

### PR DESCRIPTION
**Pull request by maniator**
update MonkeyPatched typings and removed some duplicated code

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/plugins/MonkeyPatched.ts [hunk 1]
```diff
@@ -1,36 +1,27 @@
...
-export abstract class MonkeyPatched {
-    private enabled: boolean;
+export abstract class MonkeyPatched<W> {
+    public enable = this.patch.bind(this, true);
+    public disable = this.patch.bind(this, false);
 
-    constructor() {
-        this.enabled = false;
-    }
+    protected abstract patches: MonkeyPatch<W>[];
```
Comment (maniator) [inline src/plugins/MonkeyPatched.ts:14]: This is where all the new typings are :-)

**Comment by maniator** [comment]
Hmm will fix ts issues.

Wish there was a pre-commit hook to auto-find/fix these issues better.

**Comment by maniator** [comment]
Reviewed it with @adebayor123 :-)

File: src/plugins/event-plugins/FetchPlugin.ts [hunk 2]
```diff
@@ -47,14 +47,17 @@ export const FETCH_PLUGIN_ID = 'com.amazonaws.rum.fetch';
...
+    extends MonkeyPatched<Window, 'fetch'>
+    implements Plugin {
+    private readonly pluginId: string;
+    private readonly config: HttpPluginConfig;
     private context: PluginContext;
 
     constructor(config?: PartialHttpPluginConfig) {
         super();
         this.pluginId = FETCH_PLUGIN_ID;
+        // TODO: fix config merge so that it is not shallow
```
Comment (qhanam) [inline src/plugins/event-plugins/FetchPlugin.ts:60]: nit(blocking): Remove TODOs (here and elsewhere)

**Review by qhanam — COMMENTED** [review]
Very nice.

I think this is good to go after the nit is fixed.

File: src/plugins/event-plugins/FetchPlugin.ts (hunk 2 above)
Comment (maniator) [reply src/plugins/event-plugins/FetchPlugin.ts:60]: Can do 👍🏻

**Comment by maniator** [comment]
> Very nice.
> 
> I think this is good to go after the nit is fixed.

Done @qhanam 😄

**Review by qhanam — APPROVED** [review]

**Pull request merged**